common/transport/amqp/src/amqp_message.ts

```diff
@@ -141,47 +144,59 @@ export class AmqpMessage {
 
-    if (amqpMessage.properties) {
-      /*Codes_SRS_NODE_IOTHUB_AMQPMSG_16_005: [The `toMessage` method shall set the `Message.to` property to the `AmqpMessage.properties.to` value if it is present.]*/
-      if (amqpMessage.properties.to) {
-        msg.to = amqpMessage.properties.to;
-      }
+    /*Codes_SRS_NODE_IOTHUB_AMQPMSG_16_005: [The `toMessage` method shall set the `Message.to` property to the `AmqpMessage.to` value if it is present.]*/
+    if (amqpMessage.to) {
+      msg.to = amqpMessage.to;
+    }
 
-      /*Codes_SRS_NODE_IOTHUB_AMQPMSG_16_006: [The `toMessage` method shall set the `Message.expiryTimeUtc` property to the `AmqpMessage.properties.absoluteExpiryTime` value if it is present.]*/
-      if (amqpMessage.properties.absoluteExpiryTime) {
-        msg.expiryTimeUtc = amqpMessage.properties.absoluteExpiryTime;
-      }
+    /*Codes_SRS_NODE_IOTHUB_AMQPMSG_16_006: [The `toMessage` method shall set the `Message.expiryTimeUtc` property to the `AmqpMessage.absolute_expiry_time` value if it is present.]*/
+    if (amqpMessage.absolute_expiry_time) {
+      msg.expiryTimeUtc = amqpMessage.absolute_expiry_time;
+    }
 
-      /*Codes_SRS_NODE_IOTHUB_AMQPMSG_16_004: [The `toMessage` method shall set the `Message.messageId` property to the `AmqpMessage.properties.messageId` value if it is present.]*/
-      if (amqpMessage.properties.messageId) {
-        msg.messageId = amqpMessage.properties.messageId;
+    //
+    // SERIALIZATION DEPENDENCY!!
+    // The rhea library will de-serialize an encoded uuid (0x98) as a 16 byte buffer.
+    // Since common messages should only have type string it is safe to decode
+    // these as strings.
+    //
+    /*Codes_SRS_NODE_IOTHUB_AMQPMSG_16_004: [The `toMessage` method shall set the `Message.messageId` property to the `AmqpMessage.message_id` value if it is present.]*/
+    if (amqpMessage.message_id) {
+      if (((amqpMessage.message_id as any) instanceof Buffer) && (amqpMessage.message_id.length === 16)) {
+        msg.messageId = rheaUuidToString(amqpMessage.message_id as any);
+      } else {
+        msg.messageId = amqpMessage.message_id;
       }
+    }
 
-      /*Codes_SRS_NODE_IOTHUB_AMQPMSG_16_003: [The `toMessage` method shall set the `Message.correlationId` property to the `AmqpMessage.properties.correlationId` value if it is present.]*/
-      if (amqpMessage.properties.correlationId) {
-        msg.correlationId = amqpMessage.properties.correlationId;
+    /*Codes_SRS_NODE_IOTHUB_AMQPMSG_16_003: [The `toMessage` method shall set the `Message.correlationId` property to the `AmqpMessage.correlation_id` value if it is present.]*/
+    if (amqpMessage.correlation_id) {
+      if (((amqpMessage.correlation_id as any) instanceof Buffer) && (amqpMessage.correlation_id.length === 16)) {
+        msg.correlationId = rheaUuidToString(amqpMessage.correlation_id as any);
+      } else {
+        msg.correlationId = amqpMessage.correlation_id;
       }
```

Comment:
nit: could have a helper to deserialize a GUID since this code appears to repeat at least twice.

Reply:
like maybe call it decodeUuid to mirror encodeUuid.  (or rename them to serialize/deserialize, but I like the symmetry here.)

---
In reply to: [209371927](https://github.com/Azure/azure-iot-sdk-node/pull/344#discussion_r209371927) [](ancestors = 209371927)